skbio/stats/power.py

```diff
@@ -483,6 +519,23 @@ def subsample_paired_power(test, meta, cat, control_cats, order=None,
     if num_ids <= min_observations:
         raise ValueError('There are not enough samples for subsampling.')
 
+    # Checks the ratio
+    num_groups = len(sub_ids)
+    if ratio is None:
+        ratio = np.ones((num_groups))
+    ratio = np.asarray(ratio)
+    if not ratio.shape == (num_groups,):
+        raise ValueError('There must be a ratio for each group.')
+
+    # Determines the number of p values returned by the test
+    p_return = test(sub_ids)
+    if isinstance(p_return, float):
+        num_p = 1
+    elif isinstance(p_return, np.ndarray) and len(p_return.shape) == 1:
+        num_p = p_return.shape[0]
+    else:
+        raise TypeError('test must return a float or one-dimensional array.')
+
```

Comment:
This code appears one more time, so it's probably worth refactoring into a helper function (will make it easier to keep updated/in synch).